includes/Modules/Subscribe_With_Google.php

```diff
@@ -47,7 +47,71 @@ public function register() {
 			return;
 		}
 
-		// TODO: Bring back SwG functionality after #3120 is merged.
+		// Register "access" meta field.
+		register_post_meta(
+			'',
+			'sitekit__reader_revenue__access',
+			array(
+				'show_in_rest' => true,
+				'single'       => true,
+				'type'         => 'string',
+			)
+		);
+
+		// Add "Set access to..." bulk edit option.
+		add_filter(
+			'bulk_actions-edit-post',
+			function( $bulk_actions ) {
+				$bulk_actions['sitekit-swg-access'] = __( 'Set access to...', 'google-site-kit' );
+				return $bulk_actions;
+			}
+		);
+
+		add_filter(
+			'handle_bulk_actions-edit-post',
+			function ( $redirect_to, $action, $post_ids ) {
+				if ( 'sitekit-swg-access' !== $action ) {
+					return $redirect_to;
+				}
+
+				// TODO: Add a nonce?
+				// phpcs:ignore WordPress.Security.NonceVerification, WordPress.VIP.SuperGlobalInputUsage
+				if ( isset( $_GET['sitekit-swg-access-selector'] ) ) {
+					// phpcs:ignore WordPress.Security.NonceVerification, WordPress.VIP.SuperGlobalInputUsage
+					$access = sanitize_text_field( wp_unslash( $_GET['sitekit-swg-access-selector'] ) );
+					foreach ( $post_ids as $post_id ) {
+						update_post_meta( $post_id, 'sitekit__reader_revenue__access', $access );
+					}
+				}
+
+				return $redirect_to;
+			},
+			10,
+			3
+		);
+
+		add_filter(
+			'manage_post_posts_columns',
+			function( $columns ) {
+				return array_merge( $columns, array( 'sitekit__reader_revenue__access' => __( 'Access', 'google-site-kit' ) ) );
+			}
+		);
+
+		add_action(
+			'manage_post_posts_custom_column',
+			function( $column_key, $post_id ) {
+				if ( 'sitekit__reader_revenue__access' === $column_key ) {
+					$access = get_post_meta( $post_id, 'sitekit__reader_revenue__access', true );
+					if ( $access && 'openaccess' !== $access ) {
+						echo esc_html( $access );
+					} else {
+						esc_html_e( '— Free —', 'google-site-kit' );
+					}
+				}
+			},
+			10,
+			2
+		);
```

Comment:
No technical concern with this code, but from a UX perspective I wonder if that is the most appropriate way to show the access information for a post in the table. A whole column takes a lot of space, and while users can hide columns in the UI, I think there may be more subtle ways that would also show this information closer to e.g. the post title. Especially since not every post is expected to have a special state (thinking about sites that would monetize let's say 5 out of 100 posts), I'm not sure this is worth a column.

For Idea Hub for example, we display something additional right _in_ the post title column. Maybe that could be a path.

You probably don't need to change that here unless you feel strongly about that, but I think this may get flagged from a UX perspective eventually.

Reply:
I don't feel strongly either way, just wanna build something you and the Site Kit team are happy with